grails-app/conf/UrlMappings.groovy

```diff
@@ -776,6 +776,12 @@ class UrlMappings {
             action = [GET: "listOfDemoData"]
         }
 
+        // Inventory Item
+        "/api/inventoryItem/validateLot" {
```

Comment:
Lots to dig into here ...

The first issue is the API should be plural
```
/api/inventoryItems/validateLot
```
Secondly, as @awalkowiak knows, I really don't like remote procedure calls (RPC) masquerading as REST API requests. If this is a REST API, the resource should always be a noun. 

So the next question is ... do we even need the extra verb (validateLot)? Why can't we use a GET to retrieve the inventory item associated with the product and lot number.
```
/api/products/:productId/inventoryItems/:lotNumber
```
The API returns the details, including the quantity on hand from product availability and allow the client to decide whether it's valid or not. 

If we really want to encapsulate validation rules on the API, we probably want a resource to represent the validation 
https://stackoverflow.com/questions/11676550/how-to-expose-a-validation-api-in-a-restful-way

or we can use the given resource and a different verb e.g. HEAD or OPTIONS
https://softwareengineering.stackexchange.com/questions/347148/restful-service-for-checking-if-email-is-unique

Reply:
One last thing ... if we add the InventoryItemApiController, it should extend BaseDomainApiController so that we get the benefit of the rest of the verbs.